vividus-engine/src/test/java/org/vividus/steps/ExpressionAdapterTests.java

```diff
@@ -116,14 +117,15 @@ void testSupportedExpressionNestedExpr()
     }
 
     @SuppressWarnings("unchecked")
-    @Test
-    void testNestedExpressionWithoutParams()
+    @ParameterizedTest
+    @ValueSource(strings = { EXPRESSION_KEYWORD, "targets" })
```

Comment:
could you please elaborate how these changes cover fix?

Reply:
The previous regex correctly matches nested expressions without arguments only if the number of characters in this expression is multiple of three (3+3+3...). For example, it correctly parses "story_random" or "target", but not "story_indexed" or "targets". This test checks the parsing of expressions with different number of characters in order to eliminate similar problems in the future (this is quite possible, considering regex complexity)